### PR TITLE
eco-data PowerDog Local API Binding commit based on current oh master

### PR DESCRIPTION
**Pull request by wuellueb**
Replacing pull request openhab/openhab#2729 but now based on current master, as the old pull request was based on 1.6 and rebasing to 1.7 during the pull request earlier this year ;-) did not work at all for me. PowerDogLocalAPI files are completely unchanged compared to the review by @teichsta, except for conflicting files
- `bundles/binding/pom.xml`
- `distribution/openhabhome/configurations/openhab_default.cfg`
- `distribution/pom.xml`

where conflicts are resolved and include now PowerDogLocalAPI entries. Entries were inserted after the same binding as before - due to no more activities in that part of the configuration files, there should be no resulting conflicts until the pull request is accepted this time, hopefully. Also, in
- `bundles/binding/org.openhab.binding.powerdoglocalapi/pom.xml`

the version was adapted to 1.8.3-SNAPSHOT to build testwise on top of 1.8.3 and finally to 1.9.0-SNAPSHOT to successfully do a build on top of current master.

@9037568 if you can accept this pull request instead of openhab/openhab/2729 (which also links here, so this is traceable),  I will close the old one and we can proceed here. Thanks for waiting!

### CODE REVIEW

File: bundles/binding/org.openhab.binding.powerdoglocalapi/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ org.openhab.core.types,
+ org.openhab.model.item.binding,
+ org.osgi.framework,
+ org.osgi.service.component,
+ org.osgi.service.event,
+ org.slf4j
+Export-Package: org.openhab.binding.powerdoglocalapi
+Bundle-DocURL: http://www.openhab.org
+Bundle-RequiredExecutionEnvironment: J2SE-1.5
+Service-Component: OSGI-INF/binding.xml, OSGI-INF/genericbindingprovider.xml
```
Comment (9037568) [inline bundles/binding/org.openhab.binding.powerdoglocalapi/META-INF/MANIFEST.MF]: Bundle-ActivationPolicy should be set to "lazy"

File: bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiBinding.java [hunk 2]
```diff
@@ -0,0 +1,633 @@
...
+	 * @{inheritDoc
+	 */
+	@Override
+	protected void internalReceiveUpdate(String itemName, State newState) {
+		logger.debug("internalReceiveUpdate({},{}) is called!", itemName,
+				newState);
+
+		// cycle on all available powerdogs
+		for (PowerDogLocalApiBindingProvider provider : providers) {
+			if (!provider.providesBindingFor(itemName))
```
Comment (9037568) [inline bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiBinding.java]: All "if" blocks should be enclosed in braces ( { } )

File: bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiBinding.java [hunk 3]
```diff
@@ -0,0 +1,633 @@
...
+				}
+				long age = System.currentTimeMillis() - lastUpdateTimeStamp;
+				boolean itemMayUpdate = (age >= provider
+						.getRefreshInterval(itemName));
+
+				if (itemMayUpdate) {
+					// Convert new State to PowerDog set Current_Value string
+					String value = "0";
+					if (newState instanceof OnOffType) {
+						if (newState == OnOffType.ON)
```
Comment (9037568) [inline bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiBinding.java]: All "if" blocks should be enclosed in braces ( { } )

File: bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiBinding.java [hunk 4]
```diff
@@ -0,0 +1,633 @@
...
+							value = "1";
+					} // C-like Not-Zero is True, Zero is false; Powerdog does
+						// not offer boolean for PowerAPI, so this might not be
+						// the best solution, but it is sufficient
+					else if (newState instanceof OpenClosedType) {
+						if (newState == OpenClosedType.OPEN)
```
Comment (9037568) [inline bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiBinding.java]: All "if" blocks should be enclosed in braces ( { } )

File: bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiBinding.java [hunk 5]
```diff
@@ -0,0 +1,633 @@
...
+	 * @return a {@link State} which type is inherited by the {@link TypeParser}
+	 *         or a {@link StringType} if <code>item</code> is <code>null</code>
+	 */
+	private State createState(Class<? extends Item> itemType,
+			String transformedResponse) {
+		try {
+			// Assign according to output type or cast to output type directly
+			if (itemType.isAssignableFrom(SwitchItem.class)) {
+				int value = Math.round(Float.parseFloat(transformedResponse));
+				if (value > 0)
```
Comment (9037568) [inline bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiBinding.java]: All "if" blocks should be enclosed in braces ( { } )

File: bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiBinding.java [hunk 6]
```diff
@@ -0,0 +1,633 @@
...
+					return OnOffType.ON;
+				else
```
Comment (9037568) [inline bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiBinding.java]: All "if" blocks should be enclosed in braces ( { } )

File: bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiBinding.java [hunk 7]
```diff
@@ -0,0 +1,633 @@
...
+					return OnOffType.OFF;
+			} else if (itemType.isAssignableFrom(DimmerItem.class)) {
+				return new PercentType(Math.round(Float
+						.parseFloat(transformedResponse)));
+			} else if (itemType.isAssignableFrom(ContactItem.class)) {
+				int value = Math.round(Float.parseFloat(transformedResponse));
+				if (value > 0)
```
Comment (9037568) [inline bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiBinding.java]: All "if" blocks should be enclosed in braces ( { } )

File: bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiBinding.java [hunk 8]
```diff
@@ -0,0 +1,633 @@
...
+					return OpenClosedType.OPEN;
+				else
```
Comment (9037568) [inline bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiBinding.java]: All "if" blocks should be enclosed in braces ( { } )

File: bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiGenericBindingProvider.java [hunk 9]
```diff
@@ -0,0 +1,383 @@
...
+		return "powerdoglocalapi";
+	}
+
+	/**
+	 * @{inheritDoc
+	 */
+	@Override
+	public void validateItemType(Item item, String bindingConfig)
+			throws BindingConfigParseException {
+		if (!(item instanceof SwitchItem || item instanceof DimmerItem
```
Comment (9037568) [inline bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiGenericBindingProvider.java]: All "if" blocks should be enclosed in braces ( { } )

File: bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiGenericBindingProvider.java [hunk 10]
```diff
@@ -0,0 +1,383 @@
...
+		}
+		matcher.reset();
+
+		// parse regex and extract configuration data
+		while (matcher.find()) {
+			configElement = new PowerDogLocalApiBindingConfigElement();
+			configElement.serverId = matcher.group(1);
+			configElement.valueId = matcher.group(2);
+			configElement.refreshInterval = Integer.valueOf(matcher.group(3));
+			if (matcher.group(4).isEmpty()) // group 4 is optional
```
Comment (9037568) [inline bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiGenericBindingProvider.java]: All "if" blocks should be enclosed in braces ( { } )

File: bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiGenericBindingProvider.java [hunk 11]
```diff
@@ -0,0 +1,383 @@
...
+				configElement.name = "Current_Value";
+			else
```
Comment (9037568) [inline bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiGenericBindingProvider.java]: All "if" blocks should be enclosed in braces ( { } )

File: bundles/binding/org.openhab.binding.powerdoglocalapi/META-INF/MANIFEST.MF [hunk 12]
```diff
@@ -0,0 +1,30 @@
+Manifest-Version: 1.0
+Private-Package: org.openhab.binding.powerdoglocalapi.internal
+Ignore-Package: org.openhab.binding.powerdoglocalapi.internal
+Bundle-License: http://www.eclipse.org/legal/epl-v10.html
+Bundle-Name: openHAB PowerDogLocalApi Binding
+Bundle-SymbolicName: org.openhab.binding.powerdoglocalapi
+Bundle-Vendor: openHAB.org
+Bundle-Version: 1.8.0.qualifier
```
Comment (wuellueb) [inline bundles/binding/org.openhab.binding.powerdoglocalapi/META-INF/MANIFEST.MF]: Done

File: bundles/binding/org.openhab.binding.powerdoglocalapi/META-INF/MANIFEST.MF (hunk 1 above)
Comment (wuellueb) [inline bundles/binding/org.openhab.binding.powerdoglocalapi/META-INF/MANIFEST.MF]: Bundle-ActivationPolicy: lazy added

File: bundles/binding/org.openhab.binding.powerdoglocalapi/OSGI-INF/binding.xml [hunk 13]
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+
+    Copyright (c) 2010-2015, openHAB.org and others.
```
Comment (wuellueb) [inline bundles/binding/org.openhab.binding.powerdoglocalapi/OSGI-INF/binding.xml]: Update to 2016 done

File: bundles/binding/org.openhab.binding.powerdoglocalapi/OSGI-INF/genericbindingprovider.xml [hunk 14]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+
+    Copyright (c) 2010-2015, openHAB.org and others.
```
Comment (wuellueb) [inline bundles/binding/org.openhab.binding.powerdoglocalapi/OSGI-INF/genericbindingprovider.xml]: Update to 2016 done

File: bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/PowerDogLocalApiBindingProvider.java [hunk 15]
```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright (c) 2010-2015, openHAB.org and others.
```
Comment (wuellueb) [inline bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/PowerDogLocalApiBindingProvider.java]: Update to 2016 done

File: bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/PowerDogLocalApiBindingProvider.java [hunk 16]
```diff
@@ -0,0 +1,86 @@
...
+
+import org.openhab.core.binding.BindingProvider;
+import org.openhab.core.items.Item;
+
+/**
+ * This interface is implemented by classes that can provide mapping information
+ * between openHAB items and the PowerDog Local API.
+ * 
+ * @author wuellueb
+ * @since 1.8.0
```
Comment (wuellueb) [inline bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/PowerDogLocalApiBindingProvider.java]: Update to 1.9.0 done

File: bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiBinding.java [hunk 17]
```diff
@@ -0,0 +1,633 @@
+/**
+ * Copyright (c) 2010-2015, openHAB.org and others.
```
Comment (wuellueb) [inline bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiBinding.java]: Update to 2016 done

File: bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiBinding.java [hunk 18]
```diff
@@ -0,0 +1,633 @@
...
+import org.openhab.core.types.State;
+import org.osgi.framework.BundleContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Queries eco-data PowerDog
+ * 
+ * @author wuellueb
+ * @since 1.8.0
```
Comment (wuellueb) [inline bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiBinding.java]: Update to 1.9.0 done

File: bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiBinding.java [hunk 19]
```diff
@@ -0,0 +1,633 @@
...
+					// to the config
+					String unit = provider.getServerId(itemName);
+					PowerDogLocalApiServerConfig server = serverList.get(unit);
+
+					// get the server specific update time and check if it needs
+					// an update
+					boolean serverNeedsUpdate = false;
+					if (server == null) {
+						serverNeedsUpdate = false;
+						logger.error("Unknown PowerDog server referenced: "
```
Comment (wuellueb) [inline bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiBinding.java]: Changed to parameterized logging for all logger references

File: bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiBinding.java (hunk 2 above)
Comment (wuellueb) [inline bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiBinding.java]: Done

File: bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiBinding.java (hunk 3 above)
Comment (wuellueb) [inline bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiBinding.java]: Done

File: bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiBinding.java (hunk 4 above)
Comment (wuellueb) [inline bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiBinding.java]: Done

File: bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiBinding.java [hunk 20]
```diff
@@ -0,0 +1,633 @@
...
+						PowerDog powerdog = (PowerDog) XmlRpcProxy.createProxy(
+								server.url(), "",
+								new Class[] { PowerDog.class }, false);
+						XmlRpcStruct response = powerdog.setLinearSensorDevice(
+								server.password, provider.getValueId(itemName),
+								value);
+
+						lastUpdateMap.put(itemName, System.currentTimeMillis());
+
+						logger.debug("PowerDog.setLinearSensorDevice() result: "
```
Comment (wuellueb) [inline bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiBinding.java]: Done

File: bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiBinding.java (hunk 6 above)
Comment (wuellueb) [inline bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiBinding.java]: Done

File: bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiBinding.java (hunk 7 above)
Comment (wuellueb) [inline bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiBinding.java]: Done

File: bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiBinding.java [hunk 21]
```diff
@@ -0,0 +1,633 @@
...
+						logger.debug("value: " + value);
+					}
+				} else if ("password".equals(configKey)) {
+					deviceConfig.password = value;
+				} else if ("refresh".equals(configKey)) {
+					if (StringUtils.isNotBlank(value)) {
+						// refresh cannot be lower than refresh interval
+						deviceConfig.refresh = (int) Math.max(
+								Long.parseLong(value), refreshInterval);
+						logger.debug("value: " + value);
```
Comment (wuellueb) [inline bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiBinding.java]: Done

File: bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiBinding.java [hunk 22]
```diff
@@ -0,0 +1,633 @@
...
+					}
+				} else {
+					// cannot throw new ConfigurationException(configKey,
+					// "The given PowerDogLocalApi configKey '" + configKey +
+					// "' is unknown");
+					logger.warn("The given PowerDogLocalApi configKey '"
```
Comment (wuellueb) [inline bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiBinding.java]: Done

File: bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiBinding.java [hunk 23]
```diff
@@ -0,0 +1,633 @@
...
+							+ configKey + "' is unknown");
+				}
+				logger.debug("New Server config: " + deviceConfig.toString());
```
Comment (wuellueb) [inline bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiBinding.java]: Done

File: bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiGenericBindingProvider.java (hunk 9 above)
Comment (wuellueb) [inline bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiGenericBindingProvider.java]: This one is not justified, as this is a multiline logic expression being evaluated...

File: bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiGenericBindingProvider.java [hunk 24]
```diff
@@ -0,0 +1,383 @@
...
+
+		// parse and accept binding configuration
+		if (bindingConfig != null) {
+			PowerDogLocalApiBindingConfig config = parseBindingConfig(item,
+					bindingConfig);
+			logger.debug("bindingConfig adeed (config=" + config.toString()
+					+ ")");
+			addBindingConfig(item, config);
+		} else {
+			logger.warn("bindingConfig is NULL (item=" + item
```
Comment (wuellueb) [inline bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiGenericBindingProvider.java]: Done

File: bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiGenericBindingProvider.java (hunk 10 above)
Comment (wuellueb) [inline bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiGenericBindingProvider.java]: Done

File: bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiGenericBindingProvider.java [hunk 25]
```diff
@@ -0,0 +1,383 @@
...
+		return outBindings;
+	}
+
+	/**
+	 * This is a helper class holding binding specific configuration details to
+	 * map commands to {@link PowerDogLocalAPIBindingConfigElement }. There will
+	 * be a map like <code>ON->PowerDogLocalAPIBindingConfigElement</code>
+	 * 
+	 * @author wuellueb
+	 * @since 1.8.0
```
Comment (wuellueb) [inline bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiGenericBindingProvider.java]: Done

**Comment by wuellueb** [comment]
Wiki page created and eclipse formatter re-applied (and turned on again as a default in eclipse).

**Comment by wuellueb** [comment]
[Wiki page can be found here](https://github.com/openhab/openhab/wiki/PowerDog-Local-API-binding)
Who will add it to the Wiki sidebar once the pull-request is accepted?

**Comment by 9037568** [comment]
It appears your code no longer builds because of this change to PowerDogLogalApiBinding:

image

**Comment by 9037568** [comment]
The wiki page looks good. You should probably add a note that the binding is not available before the 1.9.0 release.

You can add the page to the sidebar yourself, or I can do it for you.

**Comment by wuellueb** [comment]
Hi Chris,

you're right - I'm just now working on it. When I turned on in eclipse to do auto-formatting of the java-files, I also turned on the optimsation of the
imports... This causes that eclipse removed most of them. I'm currently running a full maven build. Once succeeded, I will commit, comment and adress you in my
comment.

Regards,
Wuellueb

Am 17.12.2016 um 14:24 schrieb Chris Carman:

**Comment by wuellueb** [comment]
Wikipage was added to the sidebar.

**Comment by wuellueb** [comment]
Hint added to wiki page that the binding is not available before the 1.9.0 release

**Comment by wuellueb** [comment]
@9037568 Now the build using mvn run's through smoothly again. Can you re-check the pull-request and pull-it in now?

**Comment by 9037568** [comment]
Build now looks good. However, you still have some files that need to be run through the Eclipse formatter:

        modified:   src/main/java/org/openhab/binding/powerdoglocalapi/PowerDogLocalApiBindingProvider.java
        modified:   src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiBinding.java
        modified:   src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiGenericBindingProvider.java

**Comment by wuellueb** [comment]
@9037568 it should be OK now - linefeeds were changed when transferred by Github desktop - which obviously was incorrect. I used TortoiseGIT for transferring now and to me it looks now that the eclipse formatter output on my disk is bitwise identical to what is on github, which was not the case before. Can you doublecheck if it is now OK?

**Comment by 9037568** [comment]
You appear to be getting tabs instead of spaces, probably indicating your formatter is not set up correctly.

````
00000950h: 2E 30 0D 0A 20 2A 2F 0D 0A 70 75 62 6C 69 63 20 ; .0.. */..public 
00000960h: 63 6C 61 73 73 20 50 6F 77 65 72 44 6F 67 4C 6F ; class PowerDogLo
00000970h: 63 61 6C 41 70 69 47 65 6E 65 72 69 63 42 69 6E ; calApiGenericBin
00000980h: 64 69 6E 67 50 72 6F 76 69 64 65 72 20 65 78 74 ; dingProvider ext
00000990h: 65 6E 64 73 0D 0A 09 09 41 62 73 74 72 61 63 74 ; ends....Abstract
````

**Comment by wuellueb** [comment]
I changed for the commit from eclipse luna w/ default settings (luna was standard eclipse when I started the OH development) to neon.1 w/ default settings. I hope it does the trick now - at least I see indenting spaces now instead of tabs...

**Comment by 9037568** [comment]
Much better. Not sure how, but looks like one got missed, though.

In PowerDogLocalApiBinding.java, on line 450, there's a stray space.

This:
>for (Iterator&lt; String>keys = keyset.iterator(); keys.hasNext();) {

needs to be changed to this:
>for (Iterator&lt;String>keys = keyset.iterator(); keys.hasNext();) {

**Comment by wuellueb** [comment]
Should be fixed now also. Eclipse auto format did not catch it...

**Comment by 9037568** [comment]
Great. Thanks, @wuellueb !

**Pull request merged**